src/daemon/plugin.c

```diff
@@ -1078,14 +1084,96 @@ static int plugin_compare_read_func(const void *arg0, const void *arg1) {
     return 0;
 } /* int plugin_compare_read_func */
 
+/*
+ * Set the time of the first read according to the settings AlignRead and
+ * Interval in collectd.conf.
+ *
+ * The AlignRead value is in seconds. The value modulo 60 determines the second
+ * of a minute. The integer part of the value divided by 60 determines the
+ * minute of an hour. Optionally, decimal places can be used to set the
+ * fraction of a second (up to microseconds precision).
+ */
+static void plugin_set_first_read_time(read_func_t *rf) {
+  double alignRead = rf->rf_ctx.align_read; /* (default is -1.0) */
+
+  /* If AlignRead is not set, start reading immediately. */
+  if (alignRead < 0) {
+    rf->rf_next_read = cdtime();
+    return;
+  }
+
+  /* AlignRead values greater than Interval do not make sense. */
+  if (alignRead >= rf->rf_interval) {
+    static bool align_error = true;
+
+    if (align_error) {
+      ERROR("AlignRead has to be smaller than Interval (%.3lf >= %ld)!",
+            alignRead, CDTIME_T_TO_TIME_T(rf->rf_interval));
+      align_error = false;
+    }
+
+    rf->rf_next_read = cdtime();
+    return;
+  }
+
+  /* Seconds and minutes of Interval and AlignRead have to be perfect divisors
+   * of 60. Otherwise to perserve alignment. */
+  int sec_interval = CDTIME_T_TO_TIME_T(rf->rf_interval) % 60;
+  int sec_align = (long)alignRead % 60;
+  int min_interval = CDTIME_T_TO_TIME_T(rf->rf_interval) / 60;
+  int min_align = (int)(alignRead / 60);
+  if ((sec_interval > 0 && 60 % sec_interval != 0) ||
+      (sec_align > 0 && 60 % sec_align != 0) ||
+      (min_interval > 0 && 60 % min_interval != 0) ||
+      (min_align > 0 && 60 % min_align != 0)) {
+    static bool align_error = true;
+
+    if (align_error) {
+      ERROR("Seconds and minutes of AlignRead and Interval have to be perfect "
+            "divisors of 60 to perserve alignment.");
+      align_error = false;
+    }
+
+    rf->rf_next_read = cdtime();
+    return;
+  }
+
+  /* get minutes in current hour and seconds in current minute */
+  cdtime_t now = cdtime();
+  struct timeval now_tv = CDTIME_T_TO_TIMEVAL(now);
+  struct tm *now_tm;
+  now_tm = localtime(&now_tv.tv_sec);
+
+  /* put current minutes, seconds and fractions of a second in a microseconds
+   * value (with stripped hours, days, etc.) */
```

Comment:
A bothering comment again: "Calculate microseconds since start of current hour"
If you want: "Calculate microseconds since start of current hour (with stripped hours, days, etc.)"

You do not put minutes into this variable. This variable holds microseconds since start of current hour.  No minutes. No seconds.

Reply:
I changed the comment.